dev/com.ibm.ws.cdi.2.0.weld/src/com/ibm/ws/cdi/impl/weld/AbstractProxyServices.java

```diff
@@ -117,6 +120,8 @@ public ClassLoader run() {
                 if (cl instanceof BundleReference) {
                     Bundle b = ((BundleReference) cl).getBundle();
                     addWeldDynamicImports(b, WELD_PACKAGES);
+                } else if (cl == null) {
+                    return CLASS_LOADER_FOR_SYSTEM_CLASSES;
```

Comment:
put some comments here... and actually I don't see a reason why this code wouldn't be kept the same for the older version too?